src/main/java/net/minecraftforge/client/model/generators/BlockModelBuilder.java

```diff
@@ -18,7 +26,231 @@
  */
 public class BlockModelBuilder extends ModelBuilder<BlockModelBuilder> {
 
-    public BlockModelBuilder(ResourceLocation outputLocation, ExistingFileHelper existingFileHelper) {
+    private final RootTransformBuilder rootTransform = new RootTransformBuilder();
+
+    public BlockModelBuilder(ResourceLocation outputLocation, ExistingFileHelper existingFileHelper)
+    {
         super(outputLocation, existingFileHelper);
     }
+
+    public RootTransformBuilder rootTransform()
+    {
+        return rootTransform;
+    }
+
+    @Override
+    public JsonObject toJson()
+    {
+        JsonObject json = super.toJson();
+
+        // Write the transform to an object
+        JsonObject transform = new JsonObject();
+
+        if (!rootTransform.translation.equals(Vector3f.ZERO))
+        {
+            transform.add("translation", writeVec3(rootTransform.translation));
+        }
+
+        if (!rootTransform.scale.equals(new Vector3f(1, 1, 1)))
+        {
+            transform.add("scale", writeVec3(rootTransform.scale));
+        }
+
+        Quaternion leftRotation = rootTransform.leftRotation;
+        if (!leftRotation.equals(Quaternion.ONE))
+        {
+            transform.add("left_rotation", writeQuaternion(leftRotation));
+        }
+
+        Quaternion rightRotation = rootTransform.rightRotation;
+        if (!rightRotation.equals(Quaternion.ONE))
+        {
+            transform.add("right_rotation", writeQuaternion(rightRotation));
+        }
```

Comment:
What is more common parlance: `left_rotation` and `right_rotation`, or `rotation` and `post_rotation`? Since the deserializer code in `TransformationHelper` accepts both pairs, we should write out with the name that is more commonly understood/known.

Reply:
Yeah I think rotation and post_rotation are more descriptive.